dusty/compiler/nginx/__init__.py

```diff
@@ -27,32 +27,32 @@ def _nginx_server_name_string(port_spec):
 def _nginx_max_file_size_string():
     return "client_max_body_size {};".format(NGINX_MAX_FILE_SIZE)
 
-def _nginx_http_spec(port_spec):
+def _nginx_http_spec(port_spec, bridge_ip):
     """This will output the nginx HTTP config string for specific port spec """
     server_string_spec = "\t server {\n"
     server_string_spec += "\t \t {}\n".format(_nginx_max_file_size_string())
     server_string_spec += "\t \t {}\n".format(_nginx_listen_string(port_spec))
     server_string_spec += "\t \t {}\n".format(_nginx_server_name_string(port_spec))
-    server_string_spec += _nginx_location_spec(port_spec)
+    server_string_spec += _nginx_location_spec(port_spec, bridge_ip)
     server_string_spec += "\t }\n"
     return server_string_spec
 
-def _nginx_stream_spec(port_spec):
+def _nginx_stream_spec(port_spec, bridge_ip):
     """This will output the nginx stream config string for specific port spec """
     server_string_spec = "\t server {\n"
     server_string_spec += "\t \t {}\n".format(_nginx_listen_string(port_spec))
-    server_string_spec += "\t \t {}\n".format(_nginx_proxy_string(port_spec))
+    server_string_spec += "\t \t {}\n".format(_nginx_proxy_string(port_spec, bridge_ip))
     server_string_spec += "\t }\n"
     return server_string_spec
 
-def get_nginx_configuration_spec(port_spec_dict):
+def get_nginx_configuration_spec(port_spec_dict, docker_bridge_ip):
```

Comment:
I end up having to actually get the IP in `commands/run` and flow it through all of these in order to avoid importing a system from a compiler. There lies circular dependency madness.